### PR TITLE
Grab bag

### PR DESCRIPTION
**Pull request by alanz**
- Target GHC 8.0.x only, getting rid of broken exhaustiveness checker tweaks
- deal with "projects" not having a cabal file, in the elisp
- ghc-tree returns the tree(s) for the requested file only

### CODE REVIEW

File: hie-ghc-tree/Haskell/Ide/GhcTreePlugin.hs [hunk 1]
```diff
@@ -11,12 +10,15 @@ module Haskell.Ide.GhcTreePlugin where
 
 import           Data.Aeson
 import qualified Data.Text as T
+-- import           Haskell.Ide.Engine.MonadFunctions (logm)
```
Comment (cocreature) [inline hie-ghc-tree/Haskell/Ide/GhcTreePlugin.hs]: You might want to remove this and also the commented line below calling `logm`.

File: stack.yaml [hunk 2]
```diff
@@ -10,6 +10,10 @@ packages:
 - hie-ghc-tree
 - hie-hare
 - hie-docs-generator
+- location:
```
Comment (cocreature) [inline stack.yaml]: Why is this necessary? I don’t see any obvious changes to the HaRe-plugin that would require this.

Comment (alanz) [inline stack.yaml]: Its for the "no-cabal" project case, needing https://github.com/alanz/HaRe/commit/48102f8899a4cd837fa0891620d5264bbf09d7fe
